### DISCUSSION TITLE
What value should a combobox inside a native form submit?

### DISCUSSION THREAD

**Discussion by stefanprobst** [adobe/react-spectrum#1672] [Ideas]
Currently, placing a `ComboBox` inside a native `<form>` will submit the `inputValue` of the text field, whether `allowsCustomValue` is set or not. I think this is because `props.name` will always be placed on the `input`.

Would it make sense to make the submit behavior dsependent on `allowsCustomValue`, i.e. when the combobox is more used like a searchable select w/ `allowsCustomValue: false` it submits `selectedKey | null`, (maybe via hidden select?), and when used more like a autosuggest w/ `allowsCustomValue: true` it submits the value of the input value?

**Comment by LFDanLu** [comment]
Hmm, I hadn't considered this before. When you blur away from the ComboBox, the `inputValue` should change appropriately depending on whether or not `allowsCustomValue` is true or not (it retains the custom value for the former and resets to the selectedKey's text for the latter), would that not be sufficient? I guess the concern is if the form is submitted if the user is currently still focused on the ComboBox?

**Reply by stefanprobst** [reply]
* yes, when the `input` is focused and the user presses enter, the current `inputValue` would be submitted
* when using `allowsCustomValue: false` to force a user to pick from the list of items, the case could be made that we'd want the `id`/`key` to be submitted, not the item label.

**Reply by LFDanLu** [reply]
Good points. I'd have to play around with adding a HiddenSelect (or something similar) to the ComboBox but I think your reasoning is correct w/ regards to the `id/key` being more important than the input value for `allowsCustomValue: false`.

**Reply by LFDanLu** [reply]
created an issue [here](https://github.com/adobe/react-spectrum/issues/1690) for tracking
